src/settings/controllers/SlidingSyncController.ts

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019, 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { logger } from "matrix-js-sdk/src/logger";
+
+import SettingController from "./SettingController";
+import PlatformPeg from "../../PlatformPeg";
+import { SettingLevel } from "../SettingLevel";
+import SdkConfig from "../../SdkConfig";
+import SettingsStore from "../SettingsStore";
+
+export default class SlidingSyncController extends SettingController {
+    public async onChange(level: SettingLevel, roomId: string, newValue: any) {
+        if (newValue) {
+            // run checks against the "proxy" to make sure it is valid BEFORE we start doing things
+            const url = SdkConfig.get().sliding_sync_proxy_url;
+            if (!url) {
+                logger.error("cannot enable sliding sync without 'sliding_sync_proxy_url' being present");
+                return;
+            }
+            try {
+                await proxyHealthCheck(url);
+            } catch (err) {
+                console.error(err);
+                // force the value to false, this is a bit ew, it would be nice if we could return
+                // a bool as to whether to go through with the change or not.
+                SettingsStore.setValue("slidingSync", roomId, level, false);
+                return;
+            }
```

Comment:
I think if people enable the proxy but it doesn't work we should let them suffer in their broken application state - we don't need to validate for this at this time.

Reply:
The problem is that this could be a security leak: if you type in the wrong proxy URL and the client does no validation then the client will send up a /sync request to that server which contains their access token. Being careful and sanity checking the endpoint first seems like an easy thing to do for an improved security posture.